modules/networking/main.tf

```diff
@@ -14,11 +14,10 @@ resource "azurerm_subnet" "private" {
   address_prefixes     = [var.network_private_subnet_cidr]
   virtual_network_name = azurerm_virtual_network.default.name
 
-  service_endpoints = [
-    "Microsoft.Sql",
-    "Microsoft.Storage",
-    "Microsoft.KeyVault"
-  ]
+  service_endpoints = concat(
+    ["Microsoft.Sql", "Microsoft.KeyVault"],
+    var.private_link ? ["Microsoft.Storage.Global"] : ["Microsoft.Storage"]
+  )
```

Comment:
I was reading up on this and was curious on if something technical drove the change. 

Azure Private Link primarily supports region-specific storage services through the creation of private endpoints. These endpoints enable secure access to Azure Storage services within a particular region by mapping a private IP address from your virtual network to the storage service, ensuring traffic does not traverse the public internet.

For example, when you create a private endpoint for Azure Storage (Microsoft.Storage), it connects to storage accounts (Blob, Table, Queue, File) in a specific region, making it possible to access these services securely and privately from within your virtual network. Each private endpoint is tied to a storage account in the same region where the endpoint is created​ ([MS Learn](https://learn.microsoft.com/en-us/azure/private-link/private-link-overview))​​ ([MS Learn](https://learn.microsoft.com/en-us/azure/private-link/private-endpoint-overview))​.

Azure Private Link does support some global services, but this functionality is not typically used with Azure Storage accounts. The global reach capability of Azure Private Link allows a virtual network in one region to connect to services behind Private Link in another region. However, this is more applicable to services like Azure SQL Database, Azure Cosmos DB, or Azure Key Vault, rather than storage accounts​ ([MS Learn](https://learn.microsoft.com/en-us/azure/azure-monitor/logs/private-link-design))​.

Reply:
This is a part issue description "Provide a mechanism for us to enable [cross-region service endpoints for Azure Storage](https://learn.microsoft.com/en-us/azure/virtual-network/virtual-network-service-endpoints-overview) for a particular deployment"

So for cross region storage access we need to add "storage.global" in private subnet.
https://github.com/wandb/terraform-azurerm-wandb/issues/57